### DISCUSSION TITLE
Disable events logging

### DISCUSSION THREAD

**Discussion by omgovich** [langfuse/langfuse#1541] [Ideas]
Hey! First of all, thanks for the great tool. We have a product with millions AI requests monthly and with Langfuse we were able to analyze costs and user behavior. So the tool is a livesaver.

Unfortunately, since we have thousands of users making hundreds AI actions each, we see that our self-hosted Langfuse database gets 3 GB bigger every day.

We are planning to implement some automatic cleanup of `input` and `output` fields for old records in `traces` and `observations` tables (is it safe?). It is a thing we probably can do ourselves, but I decided to let you know that there is a demand for this kind of feature since everyone probably needs historical data about costs and trace count only (no need to read X-month-old prompts and output bodies).

We have also noticed that over 60% of the database size is `events` table which we probably don't need at all, so my questions are:
- Is there a way to disable event tracking?
- If not, can we just wipe out this table (on schedule) without breaking the app?

Thanks!

**Comment by sliday** [comment]
Same problem here tbh. Upvoted.

**Comment by marcklingen** [comment]
Thanks for raising this. The `events` table is currently not used by any production feature but meant to help debug tricky issues. You can wipe it as often as you like. We'll add automated data retention policies for the different tables in Langfuse v3 (with the current setup it is really hard to run a cron job to do this).

Related: #1716

Proposed change: https://github.com/langfuse/langfuse/issues/1717

**Comment by marcklingen** [comment]
Released in https://github.com/langfuse/langfuse/releases/tag/v2.36.0
